orchestration/dagster_orchestration/hca_orchestration/sensors.py

```diff
@@ -0,0 +1,50 @@
+import os
+from typing import List
+
+from dagster import sensor, RunRequest, SkipReason
+
+from hca_orchestration.contrib.argo_workflows import ArgoArchivedWorkflowsClientMixin, ExtendedArgoWorkflow
+from hca_orchestration.resources.base import default_google_access_token
+
+
+class ArgoHcaImportCompletionSensor(ArgoArchivedWorkflowsClientMixin):
+    def successful_hca_import_workflows(self) -> List[ExtendedArgoWorkflow]:
+        return [
+            ExtendedArgoWorkflow(workflow, argo_url=self.argo_url, access_token=self.access_token)
+            for workflow
+            in self.list_archived_workflows()
+            if workflow.metadata.name.startswith("import-hca-total")
+            and workflow.status.phase == 'Succeeded'
+        ]
+
+    def generate_run_request(self, workflow: ExtendedArgoWorkflow) -> RunRequest:
+        inflated_workflow = workflow.inflate()
+
+        return RunRequest(
+            run_key=inflated_workflow.metadata.name,
+            run_config={
+                "solids": {
+                    "post_import_validate": {
+                        "config": {
+                            "gcp_env": os.environ.get("HCA_GCP_ENV"),
+                            "google_project_name": os.environ.get("HCA_GOOGLE_PROJECT"),
+                            "dataset_name": inflated_workflow.params_dict()['data-repo-name'].removeprefix("datarepo_"),
+                        }
+                    }
+                }
+            }
+        )
+
+
+# TODO use execution context to avoid re-scanning old workflows
+@sensor(pipeline_name="validate_egress")
```

Comment:
How often will this run?

Reply:
by default the cadence is "every thirty seconds," but we can throttle that down as needed